powersimdata/input/check.py

```diff
@@ -281,6 +280,9 @@ def _check_grid_type(grid):
     :param powersimdata.input.grid.Grid grid: a Grid instance.
     :raises TypeError: if input is not a Grid instance.
     """
+    # Import within this function to avoid circular import issue
+    from powersimdata.input.grid import Grid
+
```

Comment:
What is going on here?

Reply:
The `powersimdata.input.grid` module imports `powersimdata.input.helpers` which now imports `powersimdata.input.check` so if `powersimdata.input.check` imports `powersimdata.input.grid` in the module definition then we are in trouble.

Reply:
I don't really mind this way, but another option is to just import the module (keeping it at the top of the file) e.g. 

```
import powersimdata.input.grid as _grid
```

Reply:
I don't think it matters what we call `powersimdata.input.grid` after we import it; if we import it at all, we get circular imports.

Reply:
Difference is doing `import module` vs `from module import thing` - the former doesn't seem to execute the module contents so doesn't cause a circular import. So it's an option if we want to keep imports at the top.

Reply:
TIL! Done.

Reply:
I guess `import module` just needs to find the module, but `from module import thing` means that the module has to be inspected, which means that any imports from that module are executed?

Reply:
Nice. I prefer to have imports at the top of the module

Reply:
Actually turns out either way needs to run all the lines in the module (tested by adding print statements). But nevertheless, `import module` still fixes the circular import, not sure why..